app/static/ringing_room.js

```diff
@@ -787,6 +811,489 @@ $(document).ready(function() {
+                // {
+                //    cover: int,       // The number of rows covered by the call
+                //    every: int,       // Every how many rows the call can be called
+                //    from: int,        // The offset of calls from the lead end (e.g. for Stedman calls
+                //                         `every` = 6 and `from` = -3 or 3)
+                //    notation: string, // The place notation of the call
+                //    symbol: string    // The symbol of the call (is '-' for bobs and 's' for singles)
+                // }
+                // into what Wheatley expects (a map of indices to place notations)
+                var convert_call = function (call) {
+                    if (call === undefined) {
+                        return {};
+                    }
+                    let converted_call = {};
+                    for (let i = 0; i < method.lengthOfLead / call.every; i++) {
+                        converted_call[call.from + i * call.every] = call.notation;
+                    }
+                    return converted_call;
+                };
+
+                // Log what we're sending Wheatley for ease of debugging
+                console.log("Setting Wheatley method to " + method.title);
+
+                // Emit the socketio signal to tell Wheatley what to ring
+                socketio.emit(
+                    "c_wheatley_row_gen",
+                    {
+                        tower_id: window.tower_parameters.id,
+                        row_gen: {
+                            type: "method",
+                            title: method.title,
+                            stage: method.stage,
+                            notation: method.notation,
+                            url: method.url,
+                            bob: method.calls ? convert_call(method.calls["Bob"]) : {},
+                            single: method.calls ? convert_call(method.calls["Single"]) : {}
+                        }
+                    }
+                );
+
+                // Clear the method name box
+                this.method_name = "";
+            },
+
+            update_comp_suggestions: function (partial_comp_name) {
+                if (partial_comp_name == "") {
+                    this.current_complib_comp = undefined;
+                    this.complib_error = "Start typing a comp ID...";
+
+                    return;
+                }
+
+                if (!/^\d+$/.test(partial_comp_name)) {
+                    this.current_complib_comp = undefined;
+                    this.complib_error = "Please enter a valid number";
+
+                    return;
+                }
+
+                // Keep a reference to the correct 'this'
+                let _this = this;
+
+                let api_url = 'https://api.complib.org/composition/' + partial_comp_name;
+                let standard_url = 'https://complib.org/composition/' + partial_comp_name;
```

Comment:
I am blocked from fetching this in the background by the CORS policy on complib

![image](https://user-images.githubusercontent.com/10487726/93135708-ad57d100-f6d2-11ea-81e1-c37bc318ef8b.png)

Reply:
Yeah I'm pestering Graham John to fix this - apparently he hasn't touched CORS for CompLib which seems plausible because he doesn't do much browser stuff outside CompLib itself.

Reply:
Fixed